migrations/v0_to_v1.go

```diff
@@ -0,0 +1,91 @@
+package migrations
+
+import (
+	"context"
+	"io"
+
+	"github.com/Scalingo/go-utils/etcd"
+	"github.com/Scalingo/go-utils/logger"
+	"github.com/Scalingo/link/locker"
+	"github.com/Scalingo/link/models"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"go.etcd.io/etcd/v3/clientv3"
+)
+
+type V0toV1 struct {
+	// TODO Do we need the lease manager
+	leaseManager locker.EtcdLeaseManager
+	storage      models.Storage
+}
+
+func NewV0toV1Migration(leaseManager locker.EtcdLeaseManager, storage models.Storage) V0toV1 {
+	return V0toV1{
+		leaseManager: leaseManager,
+		storage:      storage,
+	}
+}
+
+func (m V0toV1) NeedsMigration(ctx context.Context) bool {
+	log := logger.Get(ctx)
+	log.Info("Data migration from v0 to v1 is needed")
+	// TODO
+	return false
+}
+
+func (m V0toV1) Migrate(ctx context.Context) error {
+	log := logger.Get(ctx)
+	log.Info("Migrate data from v0 to v1")
+
+	host, err := m.storage.GetCurrentHost(ctx)
+	if err != nil {
+		return errors.Wrap(err, "fail to get the current host information to migrate data from v0 to v1")
+	}
+
+	etcdClient, closer, err := newEtcdClient()
+	if err != nil {
+		return errors.Wrap(err, "fail to create etcd client to migrate data from v0 to v1")
+	}
+	defer closer.Close()
+
+	v0Storage := newV0EtcdStorage(etcdClient, clientv3.LeaseID(host.LeaseID))
```

Comment:
You should get the leaseID from the lease manager. It might not be initialized yet in the models.

Reply:
Oh sniff, I hesitated but found it prettier like that 😢